### ISSUE TITLE
unstable test in the TestHashRowContainer

### ISSUE THREAD

**Issue by hawkingrei** [pingcap/tidb#32206]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
FAIL: TestHashRowContainer (0.00s)

[2022-02-09T10:07:33.180Z]     hash_table_test.go:143: 

[2022-02-09T10:07:33.180Z]         	Error Trace:	hash_table_test.go:143

[2022-02-09T10:07:33.180Z]         	            				hash_table_test.go:95

[2022-02-09T10:07:33.180Z]         	Error:      	Not equal: 

[2022-02-09T10:07:33.180Z]         	            	expected: true

[2022-02-09T10:07:33.180Z]         	            	actual  : false

[2022-02-09T10:07:33.180Z]         	Test:       	TestHashRowContainer
```

https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/34862/pipeline

**Comment by tiancaiamao** [comment]
See also https://github.com/pingcap/tidb/issues/33749

**Comment by hawkingrei** [comment]
it is fixed by #33749
